cmd/ls/ls.go

```diff
@@ -43,12 +45,30 @@ func formatFileName(name string) string {
 	return name
 }
 
+func lookupIds(uid, gid string) (string, string) {
+	u, err := user.LookupId(uid)
+	if err != nil {
+		fatal(err)
+	}
+	g, err := user.LookupGroupId(gid)
+	if err != nil {
+		fatal(err)
+	}
+	return u.Username, g.Name
+}
+
 func printFileList(fileInfo os.FileInfo) string {
+	stat := fileInfo.Sys().(*syscall.Stat_t)
+	usr, group := lookupIds(fmt.Sprint(stat.Uid), fmt.Sprint(stat.Gid))
```

Comment:
Seems like a job to Atoi ?

https://golang.org/pkg/strconv/#Atoi

Safer too ;-)